portal/v2/src/ClusterDetailListComponents/ClusterOperator.tsx

```diff
@@ -0,0 +1,87 @@
+import { Component } from "react"
+import { Stack, Text, PivotItem, IStackItemStyles } from '@fluentui/react';
+import { ICondition, IClusterOperator } from "./ClusterOperatorsWrapper";
+import { contentStackStylesNormal } from "../App";
+import { MultiInfoList } from "./InfoList"
+
+export interface ClusterOperatorComponentProps {
+    clusterOperators: IClusterOperator[]
+    clusterName: string
+    clusterOperatorName: string
+}
+
+const stackItemStyles: IStackItemStyles = {
+    root: {
+      width: "100%",
+    },
+  };
+
+  
+  const ConditionDetails: ICondition = {
+    status: "Status",
+    reason: "Reason",
+    lastUpdated: "LastUpdated",
+    message: "Message"
+  }
+
+export interface IClusterOperatorsState {
+    clusterOperators: IClusterOperator[]
+    clusterName: string
+    clusterOperatorName: string
+}
+
+const renderClusterOperators = (operator: IClusterOperator) => {
+    return <PivotItem key={operator.name} headerText={operator.name}>
+                <Stack>
+                    <Text variant="xLarge">{operator.name}</Text>
+                    <Text variant="large" styles={contentStackStylesNormal}>Conditions</Text>
+                </Stack>
+                <Stack wrap styles={stackItemStyles}horizontal grow>
+                    <MultiInfoList headers={ConditionDetails} items={operator.conditions} title="Conditions" subProp="type" titleSize="medium"/>
+                </Stack>
+            </PivotItem>;
+};
+
+function Operators(props: {
+    clusterOperators: IClusterOperator[],
+    clusterOperatorName: string
+}) {   
+    let currentClusterOperator: IClusterOperator | null = null
+    
+    props.clusterOperators.forEach((operator: IClusterOperator) => {
+        if (operator.name === props.clusterOperatorName) {
+            currentClusterOperator = operator
+            return
+        }
+    })
+    
+    return (
+
+            <>
+                    {(currentClusterOperator != undefined) ? 
+                        renderClusterOperators(currentClusterOperator!) : null}
+            </>
+    );
```

Comment:
The `forEach` here seems like extra work to avoid just doing a [for..of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of). If you used a for..of you could immediately return the renderedClusterOperator if found, and then return null as default for the function.

Reply:
`props.clusterOperators.forEach((operator: IClusterOperator) => {
        if (operator.name === props.clusterOperatorName) {
            return (

                <>
                        {renderClusterOperators(operator)}
                </>
        );
        }
    })
    return <></>`
    
  I've managed to do this suggestion while still using foreach, is there any reason that for..of is better?

Reply:
This will just return `<></>` every time. If you do it like you've just posted, all that's happening is that you're creating an array something like `[undefined, undefined, undefined, <component>, undefined]` whose value isn't used anywhere.

Reply:
Fixed